common/src/main/java/net/adoptopenjdk/icedteaweb/logging/LoggerFactory.java

```diff
@@ -6,14 +6,28 @@
 public class LoggerFactory {
 
     private static final String FACTORY_CLASS = "net.sourceforge.jnlp.util.logging.OutputControllerLoggerFactory";
-    private static final LoggerFactoryImpl factory = initFactory();
+    private static final LoggerFactoryImpl factory;
 
-    private static LoggerFactoryImpl initFactory() {
```

Comment:
Why is this not a method anymore?

Reply:
because we cannot call LOG.xxx before we assigned the static field `factory`
So if we make this a method which returns the factory we cannot log in that method.